python_modules/dagster/dagster/_core/host_representation/external_data.py

```diff
@@ -1143,12 +1153,31 @@ def external_repository_data_from_def(
             if nested_resource.type == NestedResourceType.TOP_LEVEL:
                 inverted_nested_resources_map[nested_resource.name][resource_key] = attribute
 
-    resource_asset_usage_map = defaultdict(list)
+    resource_asset_usage_map: Dict[str, List[AssetKey]] = defaultdict(list)
     for asset in asset_graph:
         if asset.required_top_level_resources:
             for resource_key in asset.required_top_level_resources:
                 resource_asset_usage_map[resource_key].append(asset.asset_key)
 
+    # maps resource key to map of pipeline name to list of solids in that pipeline that use that resource
+    resource_job_usage_map: Dict[str, Dict[str, List[str]]] = defaultdict(lambda: defaultdict(list))
+    for pipeline in pipeline_datas or []:
+        pipeline_name = pipeline.name
+        if is_base_asset_job_name(pipeline_name):
+            continue
+        for solid_name in pipeline.pipeline_snapshot.solid_names:
+            try:
+                solid = pipeline.pipeline_snapshot.get_node_def_snap(solid_name)
```

Comment:
given that node def and job def snaps are basically already retrieved here, makes me even more dubious of creating this new JobWithOps schema